### ISSUE TITLE
User-assisted refactorings

### ISSUE THREAD

**Issue by Novakov** [icsharpcode/RefactoringEssentials#102]
There are bunch of refactorings that I'd love to see in RefactoringEssentials: Move to folder, move member up/down, extract class. All of them require some interaction with user so they cannot be simple Roslyn Code Refactorings. 

I wonder if it is doable to split these refactorings into parts - one belongs to RE and is IDE-agnostic and does real job, second belongs to specific IDE (VSIX project in case of Visual Studio) and provides information needed by first one.

**Comment by bradphelan** [comment]
I agree. These types are the main reason I still stick with Resharper. If the following features appeared then I'd probably be willing to ditch resharper. 

* Add method parameter and pull
* Safe Delete
* Pull member up/down
